scripts/build/builders/imx.py

```diff
@@ -74,30 +75,58 @@ def __init__(self,
         self.release = release
         self.app = app
 
-    def GnBuildEnv(self):
-        return {
-            'PKG_CONFIG_PATH': self.SysRootPath('IMX_SDK_ROOT') + '/sysroots/cortexa53-crypto-poky-linux/lib/aarch64-linux-gnu/pkgconfig',
-        }
-
     def GnBuildArgs(self):
+        entries = os.listdir(self.SysRootPath('IMX_SDK_ROOT'))
+        for entry in entries:
+            if re.match(r'^environment-setup-', entry):
+                env_setup_script = entry
+                break
+
+        with open(os.path.join(self.SysRootPath('IMX_SDK_ROOT'), env_setup_script), 'r') as env_setup_script_fd:
+            lines = env_setup_script_fd.readlines()
+            for line in lines:
+                line = line.strip('\n')
+                if re.match(r'export SDKTARGETSYSROOT=', line):
+                    sdk_target_sysroot = line[len(r'export SDKTARGETSYSROOT='):]
+                if re.match(r'export CC=', line):
+                    # remove the quotation marks, replace a shell env
+                    cc = line[len(r'export CC=')+1: -1]
```

Comment:
why not using regex captures like:

```python
m = re.match(r'^\s*export CC=\s*(.*)$', line)
if m:
    cc = m.group(1)    
```

This should probably be extracted in some helper since eventually the regexes may be more complicated to allow varying amount of spacing, maybe comments, maybe quotes.

Reply:
Otherwise, you probably want to do `startswith` and NOT regex

Reply:
For removing quotes, you should probably use `shlex.split()`

Reply:
This part have been modifed based on you comments.